From 385efe40982d46d008e4ea9936f781ca0ed56499 Mon Sep 17 00:00:00 2001
From: Barry <870709864@qq.com>
Date: Mon, 27 May 2019 12:31:26 +0800
Subject: [PATCH] for baidu download: range is not standardized

---
 src/HttpResponse.cc | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/HttpResponse.cc b/src/HttpResponse.cc
index 500cd64e..a443dbc7 100644
--- a/src/HttpResponse.cc
+++ b/src/HttpResponse.cc
@@ -74,6 +74,10 @@ void HttpResponse::validateResponse() const
   switch (statusCode) {
   case 200: // OK
   case 206: // Partial Content
+    // for baidu download: range is not standardized
+    if (std::string::npos != httpRequest_->getURIHost().find("www.baidu")) {
+      return;
+    }
     if (!httpHeader_->defined(HttpHeader::TRANSFER_ENCODING)) {
       // compare the received range against the requested range
       auto responseRange = httpHeader_->getRange();
-- 
2.15.0.windows.1

